ambry-tools/src/main/java/com.github.ambry.tools/admin/ConcurrencyTestTool.java

```diff
@@ -0,0 +1,971 @@
+        Callback<T> callback);
+
+    /**
+     * Deletes a blob from ambry pertaining to the {@code blobId} passed in and verifies that subsequent Get fails
+     * @param blobId the blobId the of that blob that needs to be fetched
+     * @param expectedErrorCode the expected error code for the Delete call
+     * @param futureResult the {@link FutureResult} where the result/exception will be set on completion
+     * @param callback the {@link Callback} that will be called on completion
+     */
+    void deleteAndValidate(String blobId, T expectedErrorCode, FutureResult<T> futureResult, Callback<T> callback);
+
+    /**
+     * Returns the default Error code that is expected for a Get or a Delete call
+     * @return the default error code that is expected for a Get or a Delete call
+     */
+    T getErrorCodeForNoError();
+
+    void shutdown()
+        throws InterruptedException;
+  }
+
+  /**
+   * An implementation of {@link PutGetHelper} for an Ambry Server
+   */
+  static class ServerPutGetHelper implements PutGetHelper<ServerErrorCode> {
+    private final String hostName;
+    private final int port;
+    private final ClusterMap clusterMap;
+    private final int maxBlobSize;
+    private final int minBlobSize;
+    private final boolean enableVerboseLogging;
+    private final ThreadLocalRandom localRandom;
+    private final NetworkClientUtils networkClientUtils;
+    private final AtomicInteger correlationIdGenerator = new AtomicInteger(0);
+
+    public ServerPutGetHelper(Properties properties, String hostName, int port, ClusterMap clusterMap, int maxBlobSize,
+        int minBlobSize, Boolean enableVerboseLogging)
+        throws Exception {
+      VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
+      NetworkClientFactory networkClientFactory =
+          new NetworkClientFactory(new NetworkMetrics(new MetricRegistry()), new NetworkConfig(verifiableProperties),
+              null, 20, 20, 10000, SystemTime.getInstance());
+      NetworkClient networkClient = networkClientFactory.getNetworkClient();
+      this.networkClientUtils = new NetworkClientUtils(networkClient, 50);
+      this.hostName = hostName;
+      this.port = port;
+      this.clusterMap = clusterMap;
+      this.maxBlobSize = maxBlobSize;
+      this.minBlobSize = minBlobSize;
+      this.enableVerboseLogging = enableVerboseLogging;
+      this.localRandom = ThreadLocalRandom.current();
+    }
+
+    /**
+     * {@inheritDoc}
+     * Uploads a blob to the server directly and returns the blobId along with the content
+     * @param futureResult the {@link FutureResult} where the result/exception will be set on completion
+     * @param callback the {@link Callback} that will be called on completion
+     * @return a {@link FutureResult} which will contain the result eventually
+     */
+    @Override
+    public FutureResult produce(FutureResult futureResult, Callback callback) {
```

Comment:
produce is not a good name. This is essentially putting, so why not call it the same?